### PR TITLE
[ota] Add component page, various tweaks

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

A few OTA-related tweaks as discussed on [Discord](https://discord.com/channels/429907082951524364/1253333337884590090).

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/ota.rst [hunk 1]
```diff
@@ -1,59 +1,23 @@
...
--  **on_begin** (*Optional*, :ref:`Automation<automation>`): An action to be performed when an OTA update is started.
-   See :ref:`ota-on_begin`.
--  **on_progress** (*Optional*, :ref:`Automation<automation>`): An action to be performed (approximately each second)
-   while an OTA update is in progress. See :ref:`ota-on_progress`.
--  **on_end** (*Optional*, :ref:`Automation<automation>`): An action to be performed after a successful OTA update.
-   See :ref:`ota-on_end`.
--  **on_error** (*Optional*, :ref:`Automation<automation>`): An action to be performed after a failed OTA update.
-   See :ref:`ota-on_error`.
--  **on_state_change** (*Optional*, :ref:`Automation<automation>`): An action to be performed when an OTA update state
-   change happens. See :ref:`ota-on_state_change`.
```
Comment (jesserockz) [inline components/ota.rst]: These should stay here as they are common to all platforms...

File: components/ota.rst
Comment (jesserockz) [inline components/ota.rst:1]: This should rename to `components/ota/index.rst`

File: components/ota_esphome.rst
Comment (jesserockz) [inline components/ota_esphome.rst:1]: Should be `components/ota/esphome.rst`

File: components/ota_http_request.rst
Comment (jesserockz) [inline components/ota_http_request.rst:1]: Should be `components/ota/http_request.rst`

File: index.rst [hunk 2]
```diff
@@ -196,13 +196,22 @@ Management and Monitoring
...
+Update Installation
+-------------------
+
+Install updates over-the-air (OTA).
+
+.. imgtable::
+
+    OTA Core, components/ota, system-update.svg, dark-invert
+    OTA Updates, components/ota_esphome, system-update.svg, dark-invert
+    OTA Updates via HTTP Request, components/ota_http_request, system-update.svg, dark-invert
```
Comment (jesserockz) [inline index.rst]:
```suggestion
    OTA Core, components/ota/index, system-update.svg, dark-invert
    OTA Updates, components/ota/esphome, system-update.svg, dark-invert
    OTA Updates via HTTP Request, components/ota/http_request, system-update.svg, dark-invert
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/ota.rst (hunk 1 above)
Comment (kbx81) [reply components/ota.rst]: Ugh, totally forgot to paste this section back in 🤦🏻

File: index.rst (hunk 2 above)
Comment (kbx81) [reply index.rst]: Done in [`3324ee7` (#3976)](https://github.com/esphome/esphome-docs/pull/3976/commits/3324ee72fbce7c6aa7a1b859d8ae92760134e7ae)

Comment (kbx81) [reply index.rst]: ...and [`630784e` (#3976)](https://github.com/esphome/esphome-docs/pull/3976/commits/630784edddbf31d7c66d24afdaf9a21031c2e069)

File: components/ota/index.rst [hunk 3]
```diff
@@ -154,48 +136,32 @@ will contain one of values for the ``OTAState`` enum. These values are:
...
 - :apiref:`ota/ota_component.h`
-- :doc:`/components/button/safe_mode`
-- :doc:`/components/switch/safe_mode`
-- :doc:`/components/ota_http_request`
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    *
```
Comment (jesserockz) [inline components/ota/index.rst]:
```suggestion

Platforms
---------

.. toctree::
    :maxdepth: 1
    :glob:

    *

See Also
--------

- :apiref:`ota/ota_component.h`
```

File: _redirects [hunk 4]
```diff
@@ -9,6 +9,9 @@
 
 /changelog/2022.12.4.html /changelog/2022.12.0.html 301
 /components/display/ili9341.html /components/display/ili9xxx.html 301
+/components/ota.html /components/ota/esphome.html 301
+/components/ota_esphome.html /components/ota/esphome.html 301
```
Comment (jesserockz) [inline _redirects]: This never existed before =)
```suggestion
```

Comment (kbx81) [reply _redirects]: Oh right 🤦🏻

File: components/ota/index.rst (hunk 3 above)
Comment (kbx81) [reply components/ota/index.rst]: Couldn't apply here, but done in [`aa78f73` (#3976)](https://github.com/esphome/esphome-docs/pull/3976/commits/aa78f7303b361728eced25348066e4f00a711c61)

**Review by jesserockz — APPROVED** [review]

**Pull request merged**